src/lib/CMakeLists.txt

```diff
@@ -19,6 +19,12 @@ set(
     concurrency/transaction_manager.hpp
     constant_mappings.cpp
     constant_mappings.hpp
+    constraints/abstract_table_constraint.cpp
+    constraints/abstract_table_constraint.hpp
+    constraints/lqp_unique_constraint.cpp
+    constraints/lqp_unique_constraint.hpp
+    constraints/table_key_constraint.cpp
+    constraints/table_key_constraint.hpp
```

Comment:
Reasons for top-level were:
* logical grouping
* one instead of two folders

Do you consider `lqp_unique_constraint.hpp` as part of storage? For me, that feels wrong.

Alternatively, I'd put it those two files here:
* /logical_query_plan
* /logical_query_plan/constraints (just two files, probably folder overkill?)

Table constraints should stay in
* /storage/constraints

Reply:
Do we need subfolders for this at all? I'd just place it in storage and lqp.

Reply:
Ok, I removed all of them.